src/function/vector_cast_functions.cpp

```diff
@@ -134,6 +134,51 @@ static void nestedTypesCastExecFunction(
     }
 }
 
+static void toUnionCastExecFunction(const std::vector<std::shared_ptr<common::ValueVector>>& params,
```

Comment:
As we discussed, this function doesn't work with the selSize of params > 1

Reply:
Fixed with https://github.com/kuzudb/kuzu/pull/5607/commits/13ca24e0f9710fe757b1df18289512a2818ad75f